ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -389,61 +389,64 @@ export class AptosClient {
   }
 
   /**
-   * Waits up to 10 seconds for a transaction to move past pending state.
-   * @param txnHash A hash of transaction
-   * @returns A Promise, that will resolve if transaction is accepted to the
-   * blockchain and reject if more then 10 seconds passed
+   * Wait for a transaction to move past pending state.
+   *
+   * There are 3 possible outcomes of transaction processing as pertaining to
+   * this function:
+   * 1. Transaction is processed and successfully committed to the blockchain.
+   * 2. Transaction is processed but rejected for some reason, and is therefore
+   *    not committed to the blockchain.
```

Comment:
Kinda weird since technically the error message is committed to the blockchain, but this is fine.

Reply:
I can make this clearer.